pareto/strategic_water_management/strategic_produced_water_optimization.py

```diff
@@ -5278,14 +5555,140 @@ def LogicConstraintStorageRule(model, s):
     # model.LogicConstraintStorage.pprint()
 
     def LogicConstraintTreatmentRule(model, r):
-        constraint = sum(model.vb_y_Treatment[r, j] for j in model.s_J) == 1
+        constraint = (
+            sum(
+                sum(model.vb_y_Treatment[r, b, j] for j in model.s_J) for b in model.s_B
+            )
+            == 1
+        )
+        return process_constraint(constraint)
+
+    model.LogicConstraintTreatmentAssignment = Constraint(
+        model.s_R,
+        rule=LogicConstraintTreatmentRule,
+        doc="Treatment technology assignment",
+    )
+
+    def LogicConstraintTreatmentRule2(model, r, t):
+        constraint = (
+            sum(model.v_F_Piped[r, p, t] for p in model.s_CP if model.p_RCA[r, p])
+            + sum(model.v_F_Piped[r, s, t] for s in model.s_S if model.p_RSA[r, s])
+        ) <= model.p_M_Flow * (
+            1
+            - sum(
+                sum(model.vb_y_Treatment[r, b, j] for j in model.s_J)
+                for b in model.s_B
+                if model.p_chi_DesalinaionTechnology[b]
+            )
+        )
+        return process_constraint(constraint)
 
+    # sum(model.v_F_Piped[n, p, t] for n in model.s_N if model.p_NCA[n, p])
+    model.LogicConstraintDesalinationFlow = Constraint(
+        model.s_R,
+        model.s_T,
+        rule=LogicConstraintTreatmentRule2,
+        doc="Logic constraint for flow after desalination",
+    )
+
+    def LogicConstraintTreatmentRule3(model, r, t):
+        constraint = model.v_F_WaterRemoved[r, t] <= model.p_M_Flow * sum(
+            sum(model.vb_y_Treatment[r, b, j] for j in model.s_J)
+            for b in model.s_B
+            if model.p_chi_DesalinaionTechnology[b]
+        )
         return process_constraint(constraint)
 
-    model.LogicConstraintTreatment = Constraint(
-        model.s_R, rule=LogicConstraintTreatmentRule, doc="Logic constraint treatment"
+    model.LogicConstraintNoDesalinationFlow = Constraint(
+        model.s_R,
+        model.s_T,
+        rule=LogicConstraintTreatmentRule3,
+        doc="Logic constraint for flow if not desalination",
+    )
+
+    def LogicConstraintDesalinationAssignmentRule(model, r):
+        if r in ["R01", "R03", "R06"]:
```

Comment:
these need to be generalized. I will leave this for @nareshsusarla  and @tarnold17 . Should we create a new binary parameter for "desalination site"? Is there an assumption that a single treatment site can only be desal or only be clean brine. Or is it possible for a single site to be one or the either?

Reply:
Addressed in my latest commits. Someone else should review the relevant changes I made and make sure they look okay.

Reply:
The new variable p_chi_desalinationSites, renaming of v_F_WaterRemoved, and edits to LogicConstraintDesalinationAssignmentRules look good to me!